[products] Show Identifiers in product pages

captn3m0 opened the pull request:
Currently, we don't expose identifiers beyond storing them in the front-matter. This will start exposing the identifiers on the website as a start.

Simple version to start with, based on https://github.com/package-url/packageurl-python/blob/main/src/packageurl/contrib/purl2url.py, with a few more types, and no support for "versions", since identifiers won't have that.

captn3m0 committed:
[products] Show Identifiers in product pages

Simple version to start with.
We can look at a PURL->URL generator later

marcwrobel commented on Gemfile.lock:
Updates will be taken care of separately in #5415.

marcwrobel commented on products/alpinelinux.md:
```diff
@@ -14,6 +14,7 @@ releaseDateColumn: true
 identifiers:
 -   cpe: cpe:/o:alpinelinux:alpine_linux
 -   cpe: cpe:2.3:o:alpinelinux:alpine_linux
+-   purl: pkg:swid/alpine?tag_id=alpine
```
Update moved to #5416 to make this PR smaller.

marcwrobel commented on products/antixlinux.md:
```diff
@@ -13,9 +13,6 @@ releaseLabel: "__RELEASE_CYCLE__ (__CODENAME__)"
 eoasColumn: true
 releaseDateColumn: true
 
-identifiers:
```
Moved to #5417

marcwrobel commented on products/bellsoft-liberica.md:
```diff
@@ -358,68 +358,68 @@ identifiers:
 -   purl: pkg:apk/alpine/bellsoft-java8-runtime-lite?repository_url=https://apk.bell-sw.com/main
 
 # RPM packages that work across OpenSUSE/CentOS etc
--   purl: pkg/rpm/bellsoft-java11?repository_url=https://yum.bell-sw.com
```
Moved to #5418.

marcwrobel commented on products/cfengine.md:
```diff
@@ -10,7 +10,7 @@ eolColumn: Supported
 identifiers:
 -   repology: cfengine
--   purl: pkg:homebrew/cfengine
+-   purl: pkg:brew/cfengine
```
Moved to #5419

marcwrobel commented on products/devuan.md:
```diff
@@ -8,9 +8,6 @@ releasePolicyLink: https://www.devuan.org/os/releases
 releaseLabel: "__RELEASE_CYCLE__ (__CODENAME__)"
 releaseDateColumn: true
 
-identifiers:
```
Moved to #5420

marcwrobel commented on products/dotnet.md:
```diff
@@ -29,13 +29,13 @@ identifiers:
--   purl: pkg:rpm/rhel/dotnet-sdk-3.1
--   purl: pkg:rpm/rhel/dotnet-sdk-3.0
--   purl: pkg:rpm/rhel/dotnet-sdk-2.1
+-   purl: pkg:rpm/redhat/dotnet-sdk-8.0
```
Moved to #5421

marcwrobel commented on products/mariadb.md:
```diff
@@ -17,7 +17,7 @@ identifiers:
 -   purl: pkg:docker/bitnami/mariadb
 -   purl: pkg:docker/bitnami/mariadb-galera
--   purl: pkg:homebrew/mariadb
+-   purl: pkg:brew/mariadb
```
Moved to #5422

marcwrobel commented on products/mxlinux.md:
```diff
@@ -14,9 +14,6 @@ releaseLabel: "__RELEASE_CYCLE__ (__CODENAME__)"
 eoasColumn: true
 releaseDateColumn: true
 
-identifiers:
```
Moved to #5423

marcwrobel commented on products/mysql.md:
```diff
@@ -29,7 +29,6 @@ identifiers:
 -   repology: mysql
 -   purl: pkg:generic/mysql
 -   purl: pkg:docker/library/mysql
--   purl: pkg:binary/mysql
```
Moved to #5424.

marcwrobel commented on products/nginx.md:
```diff
@@ -11,7 +11,7 @@ releaseDateColumn: true
 identifiers:
 -   repology: nginx
--   purl: pkg:binary/nginx
+-   purl: pkg:generic/nginx
```
Moved to #5425.

marcwrobel commented on products/node-feature-discovery.md:
```diff
@@ -23,7 +23,7 @@ auto:
 identifiers:
 -   purl: pkg:github/kubernetes-sigs/node-feature-discovery
--   purl: oci:node-feature-discovery?repository_url=registry.k8s.io/nfd
+-   purl: pkg:oci/node-feature-discovery?repository_url=registry.k8s.io/nfd
```
Moved to #5426

marcwrobel commented on products/redhat-build-of-openjdk.md:
```diff
@@ -23,22 +23,22 @@ releaseLabel: "OpenJDK __RELEASE_CYCLE__"
 # This is not great, since it is quite impossible to match
 # identifiers to a release cycle.
 identifiers:
-    - purl: pkg:redhat/java-1.6.0-openjdk
```
Moved to #5427

marcwrobel commented on products/vmware-photon.md:
```diff
@@ -20,7 +20,7 @@ customColumns:
 
 identifiers:
--   purl: pkg:os/photon
+-   cpe: cpe:2.3:o:vmware:photon_os
```
Moved to #5428

marcwrobel commented on _includes/identifiers.html:
```diff
@@ -0,0 +1,32 @@
+<details>
+  <summary>
+    Show Product Identififers
```
```suggestion
    Show Product Identifiers
```

marcwrobel reviewed (commented):
Overall LGTM.

Before merging I will:

- add a validator to validate all identifiers
- see if I should exclude the validation of those generated URLs from product-data-validator
- see if I can improve the rendering (having a " Show Product Identifiers" at the  bottom seems a bit strange).

marcwrobel committed:
Various improvements

- add validation for identifiers in product-data-validator.rb (makes errors easier to read)
- simplify and generify identifiers.html (render all identifiers the same way)
- make purl-to-url generate URLs for all identifier types (rename to identifier-to-url)
- make identifier-to-url more strict for unknown identifier types or variants
- revert sass-embedded version to avoid lots of build warnings
